src/components/Scorecards/ScorecardDetailsPage/ScorecardsTableCard/ScorecardsTableCard.tsx

```diff
@@ -13,43 +13,105 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React from 'react';
+import React, { useMemo } from 'react';
+import { EmptyState, InfoCard, Table as BSTable, TableColumn } from '@backstage/core-components';
 import { ScorecardServiceScore } from '../../../../api/types';
 import { useDetailCardStyles } from '../../../../styles/styles';
-import { Table, TableBody } from '@material-ui/core';
-import { ScorecardsTableRow } from './ScorecardsTableRow';
-import { EmptyState, InfoCard } from '@backstage/core-components';
+import { Gauge } from '../../../Gauge';
+import { humanizeAnyEntityRef } from '../../../../utils/types';
+import { defaultComponentRefContext } from '../../../../utils/ComponentUtils';
+import { ScorecardLadderLevelBadge } from '../../../Common/ScorecardLadderLevelBadge';
+import { ScorecardServiceRefLink } from '../../../ScorecardServiceRefLink/ScorecardServiceRefLink';
 
 interface ScorecardsTableProps {
   scorecardId: number;
   scores: ScorecardServiceScore[];
 }
 
+const PAGE_SIZE = 25;
```

Comment:
I chose 25 somewhat arbitrarily -- feel free to suggest a different size if you think it makes more sense.